vllm/v1/core/sched/scheduler.py

```diff
@@ -500,6 +505,9 @@ def _try_schedule_encoder_inputs(
         limitations, the method adjusts `num_new_tokens` to schedule only the
         decoder tokens up to just before the unschedulable encoder input.
         """
+        if num_new_tokens == 0 or not request.has_encoder_inputs:
+            return [], num_new_tokens, encoder_budget
```

Comment:
Just wanted to double check: This is just an extra safe guard that is orthogonal to this PR, right?

Reply:
Not really. We may have requests with num_new_tokens=0 if all tokens are already scheduled.